indico/modules/rb/models/reservation_occurrences.py

```diff
@@ -250,9 +251,17 @@ def can_reject(self, user, allow_admin=True):
         return self.reservation.can_reject(user, allow_admin=allow_admin)
 
     def can_cancel(self, user, allow_admin=True):
```

Comment:
Shouldn't this also take the grace period into account? Also, below you return True in some cases without checking the false cases first (you can never cancel an occurrence if the whole booking is already cancelled). FWIW, cancelling the whole booking cancels all occurrences, so this isn't really an issue, but I think we should still change the order of those checks.